app.py

```diff
@@ -1,7 +1,22 @@
-from flask_api import FlaskAPI
+from flask_api import FlaskAPI 
+from flask import request
 
 app = FlaskAPI(__name__)
 
+RESULT = {
+        's1' : 'value1',
+        's2' : 'value2',
+        's3' : 'value3',
+        's4' : 'value4',
+        's5' : 'value5'
+}
+#  quesion 1 finshed  
+@app.route('/data', methods=['GET'])
+def query_string():
+    args = request.args['key']       
```

Comment:
What if such key does not exist in the dict?

Reply:
In fact, RESULT dict has no key named `key` at all.

Reply:
em hiểu là nếu vào trang có địa chỉ là //server-adress/data?key=s1 thì sẽ trả về giá trị của key là 's1' trong từ điển, anh nói là Query string có dạng là ?q=fhawohfo trong đó q là first parameter, thì bài này bạn ý cho nó là 'key', anh có quy định gì về parameter này là gì ko ạ, hay ý anh là code theo cách khác

Reply:
> em hiểu là nếu vào trang có địa chỉ là //server-adress/data?key=s1 thì sẽ trả về giá trị của key là 's1' trong từ điển, anh nói là Query string có dạng là ?q=fhawohfo trong đó q là first parameter, thì bài này bạn ý cho nó là 'key', anh có quy định gì về parameter này là gì ko ạ, hay ý anh là code theo cách khác

à uh anh nhìn nhầm.
Nhưng thế thì phải handle case là key được request ko tồn tại

Reply:
chỉ cần thêm viết là request.args.get('key') là dc phải không ạ, nếu key ko tồn tại nó trả về none